test/regressions/regression-319-test.coffee

```diff
@@ -0,0 +1,185 @@
+
+  cli = exec(cmd, (err, stdout, stderr) ->
+    result.exitStatus = err?.code or null
+    result.stdout = '' + stdout
+    result.stderr = '' + stderr
+  )
+  cli.on('close', (code) ->
+    result.exitStatus ?= code if code
+    cb(null, result)
+  )
+
+
+parseOutput = (output) ->
+  # Parse individual entries (deals also with multi-line entries)
+  entries = []
+  entry = undefined
+  for line in output.split(/\r?\n/)
+    match = line.match(/^(\w+): (.+)?$/)
+    if match
+      if entry
+        entry.body = entry.body.trim()
+        entries.push(entry)
+      entry = {label: match[1], body: match[2] or ''}
+    else
+      entry.body += "\n#{line.trim()}"
+
+  # Re-arrange data from entries
+  results = {summary: '', failures: [], bodies: [], schemas: []}
+  for entry in entries
+    switch entry.label
+      when 'body' then results.bodies.push(JSON.parse(entry.body) if entry.body)
+      when 'bodySchema' then results.schemas.push(JSON.parse(entry.body) if entry.body)
+      when 'complete' then results.summary = entry.body
+      when 'fail' then results.failures.push(entry.body)
+  return results
+
+
+describe('Regression: Issue #319', ->
+  results = undefined
+
+  brickTypePayload =
+    id: ''
+    name: ''
+    colors: ['red', 'brown']
+    dimensions: [[20, 30, 40]]
+
+  brickTypeSchema =
+    $schema: 'http://json-schema.org/draft-04/schema#'
+    type: 'object'
+    properties:
+      id: {type: 'string'}
+      name: {type: 'string'}
+      colors: {type: 'array'}
+      dimensions: {type: 'array'}
+
+  userPayload =
+    id: ''
+    name: ''
+    shoeSize: 42
+
+  userSchema =
+    $schema: 'http://json-schema.org/draft-04/schema#'
+    type: 'object'
+    properties:
+      id: {type: 'string'}
+      name: {type: 'string'}
+      shoeSize: {type: 'number'}
+
+  userArrayPayload = [
+    userPayload
+  ]
+
+  userArraySchema =
+    $schema: 'http://json-schema.org/draft-04/schema#'
+    type: 'array'
+
+  before((done) ->
+    app = express()
+
+    # Attaching endpoint for each testing scenario
+    app.get('/bricks/XYZ42', (req, res) ->
+      res.setHeader('Content-Type', 'application/json')
+      res.status(200).send(brickTypePayload)
+    )
+    app.post('/bricks', (req, res) ->
+      res.setHeader('Content-Type', 'application/json')
+      res.status(200).send(brickTypePayload)
+    )
+    app.get('/customers', (req, res) ->
+      res.setHeader('Content-Type', 'application/json')
+      res.status(200).send(userArrayPayload)
+    )
+    app.post('/customers', (req, res) ->
+      res.setHeader('Content-Type', 'application/json')
+      res.status(200).send(userPayload)
+    )
+
+    # Spinning up the Express server, running Dredd, and saving results
+    server = app.listen(PORT, ->
+      runDredd('./test/fixtures/regression-319.apib', (err, result) ->
+        results = parseOutput(result.stdout)
+        server.close(done)
+      )
+    )
+  )
+
+  it('outputs no failures', ->
```

Comment:
I would love to see an expectation for Dredd failing scenario when the backend doesn't reply with the expected response (missing required response key or different value type)